pylabrobot/thermocycling/thermocycler.py

```diff
@@ -0,0 +1,253 @@
+          lid_temperature: Lid temperature to set during the profile.
+          pre_denaturation_temp: Optional pre-denaturation temperature in °C.
+          pre_denaturation_time: Optional pre-denaturation time in seconds.
+          final_extension_temp: Optional final extension temperature in °C.
+          final_extension_time: Optional final extension time in seconds.
+          storage_temp: Optional storage temperature in °C.
+          storage_time: Optional storage time in seconds.
+        """
+
+        await self.set_lid_temperature(lid_temperature)
+        await self.wait_for_lid()
+
+        profile = []
+
+        if pre_denaturation_temp is not None and pre_denaturation_time is not None:
+            profile.append({"celsius": pre_denaturation_temp, "holdSeconds": pre_denaturation_time})
+
+        # Main PCR cycles
+        pcr_step = [
+            {"celsius": denaturation_temp, "holdSeconds": denaturation_time},
+            {"celsius": annealing_temp, "holdSeconds": annealing_time},
+            {"celsius": extension_temp, "holdSeconds": extension_time},
+        ]
+        for _ in range(num_cycles):
+            profile.extend(pcr_step)
+
+        if final_extension_temp is not None and final_extension_time is not None:
+            profile.append({"celsius": final_extension_temp, "holdSeconds": final_extension_time})
+
+        if storage_temp is not None and storage_time is not None:
+            profile.append({"celsius": storage_temp, "holdSeconds": storage_time})
+
+        return await self.run_profile(
+            profile=profile,
+            block_max_volume=block_max_volume
+        )
+
+    async def get_block_current_temperature(self) -> float:
+        """Get the current block temperature (°C)."""
+        return await self.backend.get_block_current_temperature()
+
+    async def get_block_target_temperature(self) -> Optional[float]:
+        """Get the block’s target temperature (°C)."""
+        return cast(Optional[float], await self.backend.get_block_target_temperature())
+
+    async def get_lid_current_temperature(self) -> float:
+        """Get the current lid temperature (°C)."""
+        return await self.backend.get_lid_current_temperature()
+
+    async def get_lid_target_temperature(self) -> Optional[float]:
+        """Get the lid’s target temperature (°C), if supported."""
+        return await self.backend.get_lid_target_temperature()
+
+    async def get_lid_status(self) -> str:
+        """Get whether the lid is “open” or “closed”."""
+        return cast(str, await self.backend.get_lid_status())
+
+    async def get_hold_time(self) -> float:
+        """Get remaining hold time (s) for the current step."""
+        return await self.backend.get_hold_time()
+
+    async def get_current_cycle_index(self) -> int:
+        """Get the zero-based index of the current cycle."""
+        return (await self.backend.get_current_cycle_index()) - 1
```

Comment:
why not do this conversion in the backend and make the convention that backends return 0-indexed cycles?